admin/class-yoast-network-admin.php

```diff
@@ -0,0 +1,248 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package WPSEO\Internals
+ */
+
+/**
+ * Multisite utility class for network admin functionality.
+ */
+class Yoast_Network_Admin {
+
+	/**
+	 * Action identifier for updating plugin network options.
+	 */
+	const UPDATE_OPTIONS_ACTION = 'yoast_handle_network_options';
+
+	/**
+	 * Action identifier for restoring a site.
+	 */
+	const RESTORE_SITE_ACTION = 'yoast_restore_site';
+
+	/**
+	 * @var bool Internal flag for whether the necessary hooks have been added.
+	 */
+	private $hooks_registered = false;
+
+	/**
+	 * @var Yoast_Network_Admin The singleton instance of this class.
+	 */
+	private static $instance = null;
```

Comment:
Perhaps we should get rid of the singleton principle, because it hard to test and hard to maintain.

Reply:
I agree that it is a bad principle, however with the current state of the codebase it's a necessary evil. I think getting rid of singletons should be tackled at a broader scope with introduction of a service container of some sort.

About testability, that is only an issue if you make the constructor protected or private, which I intentionally didn't here.

If you agree, I'd rather keep it as is for this PR, and we can discuss changes in a dedicated issue that affects the whole codebase in a consistent way.

Reply:
Looking at the code again, I think in case of the `Yoast_Network_Admin` class, the singleton is indeed not necessary since it doesn't contain functionality where we need to make sure we call it on a specific instance. So I will get rid of the singleton there.

With `Yoast_Network_Settings_API` for example it is different, here we need to ensure we use the same instance. For such cases, I remain with my previous comment that we should open an issue to find a way to globally get rid of singleton usage.